src/main/resources/static/js/grid.js

```diff
@@ -0,0 +1,92 @@
+function Cell(opt) {
+    this.el = $('<td/>');
+
+    this.percent = opt.percent;
+    this.row = opt.row;
+    this.col = opt.col;
+    this.states = opt.states;
+    this.parent = opt.parent;
+    this.state = 0;
+
+    this.setState = function(state) {
+        this.el.removeClass(this.states[this.state]);
+        this.state = state;
+        this.el.addClass(this.states[this.state]);
+    };
+
+    this.el.on('click', {cell: this}, function(e) {
+        cell = e.data.cell;
```

Comment:
The following on click function implements the state functionality. Currently, I removed the radio buttons and just cycle through the various states. This was the easiest to implement and we can change it later if necessary